controllers/volumesnapshotbackup_controller.go

```diff
@@ -94,15 +98,59 @@ func (r *VolumeSnapshotBackupReconciler) Reconcile(ctx context.Context, req ctrl
 		Name:      vsb.Name,
 	}
 
+	if VSBBatchNumber == 0 {
+		batchValue, err := GetBackupBatchValue(vsb.Spec.ProtectedNamespace, r.Client)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		VSBBatchNumber, _ = strconv.Atoi(batchValue)
+	}
+
 	// stop reconciling on this resource when completed or failed
 	if vsb.Status.Phase == volsnapmoverv1alpha1.SnapMoverBackupPhaseCompleted ||
 		vsb.Status.Phase == volsnapmoverv1alpha1.SnapMoverBackupPhaseFailed ||
 		vsb.Status.Phase == volsnapmoverv1alpha1.SnapMoverBackupPhasePartiallyFailed {
+
+		// remove from queue
 		return ctrl.Result{
 			Requeue: false,
 		}, nil
 	}
 
+	// another VSB can be added to processing batch
+	if len(vsb.Status.BatchingStatus) > 0 && vsb.Status.BatchingStatus == volsnapmoverv1alpha1.SnapMoverBackupBatchingCompleted {
+		processingVSBs--
+	}
+
+	// update non-processed VSB as queued
+	if processingVSBs >= VSBBatchNumber && len(vsb.Status.BatchingStatus) == 0 {
+		r.Log.Info(fmt.Sprintf("marking vsb %v batching status as queued", vsb.Name))
+
+		vsb.Status.BatchingStatus = volsnapmoverv1alpha1.SnapMoverBackupBatchingQueued
+		err := r.Status().Update(context.Background(), &vsb)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// requeue VSB if max batch number is still being processed
+	} else if processingVSBs >= VSBBatchNumber && vsb.Status.BatchingStatus == volsnapmoverv1alpha1.SnapMoverBackupBatchingQueued {
+		return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
```

Comment:
Please log something here.. I think I was stuck here requeuing without any updates for one eternity.

Reply:
due to VSB deletion thus processingVSBs never getting decremeted.

I believe this can be fixed by finalizer and deletionTimestamp processing.